extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -425,12 +431,18 @@ export class AmpA4A extends AMP.BaseElement {
           });
         });
 
-    this.a4aAnalyticsConfig_ = this.getA4aAnalyticsConfig();
-    if (this.a4aAnalyticsConfig_) {
-      // TODO(warrengm): Consider having page-level singletons for networks that
-      // use the same config for all ads.
-      this.a4aAnalyticsElement_ = insertAnalyticsElement(
-          this.element, this.a4aAnalyticsConfig_, true /* loadAnalytics */);
+    const networkType = this.element.getAttribute('type');
+    if (!(networkType in perNetworkAnalyticsElements)) {
+      const analyticsConfig = this.getA4aAnalyticsNetworkConfig();
+      if (analyticsConfig) {
+        this.hasA4aAnalyticsElement_ = true;
+        // NOTE(warrengm): We trigger the event to body because the analytics
+        // instrumentation service is scoped by ampdoc.
+        perNetworkAnalyticsElements[networkType] = insertAnalyticsElement(
+            this.win.document.body, analyticsConfig, true /* loadAnalytics */);
+      } else {
+        perNetworkAnalyticsElements[networkType] = null;
```

Comment:
Any reason we need to set the value to null rather than leave it undefined?

Reply:
This is done to ensure that track whether getA4aAnalyticsNetworkConfig has been called to enforce that it's called only once per page.

I added a comment and a test.

Reply:
Got it, thanks.